api/envoy/extensions/http/header_validators/envoy_default/v3/header_validator.proto

```diff
@@ -0,0 +1,112 @@
+syntax = "proto3";
+
+package envoy.extensions.http.header_validators.envoy_default.v3;
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.http.header_validators.envoy_default.v3";
+option java_outer_classname = "HeaderValidatorProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/http/header_validators/envoy_default/v3;envoy_defaultv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Envoy's default Header Validator config]
+
+// This extension validates that HTTP request and response headers are well formed according to respective RFCs.
+// 1. HTTP/1 header map validity according to `RFC 7230 section 3.2 <https://tools.ietf.org/html/rfc7230#section-3.2>`_
+// 2. Syntax of HTTP/1 request target URI and response status
+// 3. H/2 header map validity according to `RFC 7540 section 8.1.2 <https://tools.ietf.org/html/rfc7540#section-8.1.2>`_
+// 4. Syntax of HTTP/2 pseudo headers
+// 5. HTTP/3 header map validity (TBD standard link when approved)
+// 6. Syntax of HTTP/3 pseudo headers
+// 7. Syntax of Content-Length and Transfer-Encoding
+// 8. Validation of HTTP/1 requests with both Content-Length and Transfer-Encoding headers
+// 9. Normalization of the URI path according to `Normalization and Comparison <https://tools.ietf.org/html/rfc3986#section-6>`_
+// without `case normalization <https://tools.ietf.org/html/rfc3986#section-6.2.2.1>`_
+//
+message HeaderValidatorConfig {
+  message UriPathNormalizationOptions {
+    // Determines the action for request that contain %2F, %2f, %5C or %5c sequences in the URI path.
+    // This operation occurs before URL normalization and the merge slashes transformations if they were enabled.
+    enum PathWithEscapedSlashesAction {
+      // Default behavior specific to implementation (i.e. Envoy) of this configuration option.
+      // Envoy, by default, takes the KEEP_UNCHANGED action.
+      // NOTE: the implementation may change the default behavior at-will.
+      IMPLEMENTATION_SPECIFIC_DEFAULT = 0;
+
+      // Keep escaped slashes.
+      KEEP_UNCHANGED = 1;
+
+      // Reject client request with the 400 status. gRPC requests will be rejected with the INTERNAL (13) error code.
+      // The "httpN.downstream_rq_failed_path_normalization" counter is incremented for each rejected request.
+      REJECT_REQUEST = 2;
+
+      // Unescape %2F and %5C sequences and redirect request to the new path if these sequences were present.
+      // Redirect occurs after path normalization and merge slashes transformations if they were configured.
```

Comment:
redirects occur
The redirect will occur

Reply:
done